### PR TITLE
new conda package for hatch-ci

### PR DESCRIPTION
**Pull request by cav71**
Checklist
- [x] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml".
- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/5eddbd7fc9d1502169089da06c3688d9759be978/recipes/example/meta.yaml#L64-L73) for an example).
- [x] Source is from official source.
- [x] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged).
- [x] If static libraries are linked in, the license of the static library is packaged.
- [x] Package does not ship static libraries. If static libraries are needed, [follow CFEP-18](https://github.com/conda-forge/cfep/blob/main/cfep-18.md).
- [x] Build number is 0.
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details).
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there.
- [x] When in trouble, please check our [knowledge base documentation](https://conda-forge.org/docs/maintainer/knowledge_base.html) before pinging a team.

### CODE REVIEW

**Comment by cav71** [comment]
@conda-forge/help-python, ready for review!

File: LICENSE.txt [hunk 1]
```diff
@@ -1,27 +1,9 @@
-Copyright (c) 2015-2023, conda-forge
```
Comment (ocefpaf) [inline LICENSE.txt:1]: Please don't change our license!

File: recipes/hatch-ci/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+
+build:
+  script: {{ PYTHON }} -m pip install . -vv
+  number: 0
+
+requirements:
+  build:
+    # If your project compiles code (such as a C extension) then add the required compilers as separate entries here.
+    # Compilers are named 'c', 'cxx' and 'fortran'.
+    - {{ compiler('c') }}
```
Comment (ocefpaf) [inline recipes/hatch-ci/meta.yaml:23]: This looks like a noarch package. Do you need a compiler?

File: recipes/hatch-ci/meta.yaml [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+    - typing-extensions
+    - jinja2
+  run:
+    - python
+    - hatchling
+    - typing-extensions
+    - jinja2
+
+test:
+  # Some packages might need a `test/commands` key to check CLI.
```
Comment (ocefpaf) [inline recipes/hatch-ci/meta.yaml:37]: Please remove the comments from the example recipe.

File: recipes/hatch-ci/meta.yaml [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+
+source:
+  url: https://pypi.io/packages/source/{{ name[0] }}/{{ name }}/{{ name.replace("-", "_") }}-{{ version }}.tar.gz
+  sha256: 846efdcc9f2adaa2da9074e1901ce100ec3830143349ba1f5befa911b6a39a3c
+  patches:
+    # this is a one-off patch to allow to build the package (because it is a recursive dependecy)
+    - conda.patch    
+
+build:
+  script: {{ PYTHON }} -m pip install . -vv
```
Comment (ocefpaf) [inline recipes/hatch-ci/meta.yaml:16]:
```suggestion
  entry_points:
    - hatch-ci = hatch_ci:script.main
  noarch: python
  script: {{ PYTHON }} -m pip install . -vv
```

File: recipes/hatch-ci/meta.yaml [hunk 5]
```diff
@@ -0,0 +1,62 @@
...
+    - python
+    - pip
+    - hatchling
+    - typing-extensions
+    - jinja2
+  run:
+    - python
+    - hatchling
+    - typing-extensions
+    - jinja2
```
Comment (ocefpaf) [inline recipes/hatch-ci/meta.yaml:34]:
```suggestion
    - python >=3.8
    - pip
    - hatchling >=1.1.0
    - typing-extensions
    - jinja2
  run:
    - python >=3.8
    - hatchling >=1.1.0
    - typing-extensions
    - jinja2
```

File: recipes/hatch-ci/meta.yaml [hunk 6]
```diff
@@ -0,0 +1,62 @@
...
+test:
+  # Some packages might need a `test/commands` key to check CLI.
+  # List all the packages/modules that `run_test.py` imports.
+  imports:
+    - hatch_ci
+    - hatch_ci.script
+  requires:
+    - pip
+  commands:
+    - pip check
```
Comment (ocefpaf) [inline recipes/hatch-ci/meta.yaml:45]:
```suggestion
    - pip check
    - hatch-ci --help
```

File: recipes/hatch-ci/meta.yaml [hunk 7]
```diff
@@ -0,0 +1,62 @@
...
+  home: https://github.com/cav71/hatch-ci
+  summary: 'hatch plugin to leverage github for releases'
+  description: |
+    This provides a plugin to Hatch leveraging a CI/CD system (github at the moment) to deliver packages to PyPi.
+  license: MIT
+  license_family: MIT
+  # It is required to include a license file in the package,
+  # (even if the license doesn't require it) using the license_file entry.
+  # Please also note that some projects have multiple license files which all need to be added using a valid yaml list.
+  # See https://docs.conda.io/projects/conda-build/en/latest/resources/define-metadata.html#license-file
```
Comment (ocefpaf) [inline recipes/hatch-ci/meta.yaml:57]:
```suggestion
```

File: recipes/hatch-ci/meta.yaml [hunk 8]
```diff
@@ -0,0 +1,58 @@
...
+  sha256: 846efdcc9f2adaa2da9074e1901ce100ec3830143349ba1f5befa911b6a39a3c
+  patches:
+    # this is a one-off patch to allow to build the package (because it is a recursive dependecy)
+    - conda.patch    
+
+build:
+  entry_points:
+    - hatch-ci = hatch_ci:script.main
+  noarch: python
+  skip: True  # [py<37]
```
Comment (ocefpaf) [inline recipes/hatch-ci/meta.yaml:19]:
```suggestion
```

File: LICENSE.txt (hunk 1 above)
Comment (cav71) [reply LICENSE.txt]: reverted

File: recipes/hatch-ci/meta.yaml (hunk 3 above)
Comment (cav71) [reply recipes/hatch-ci/meta.yaml]: done

File: recipes/hatch-ci/meta.yaml (hunk 5 above)
Comment (cav71) [reply recipes/hatch-ci/meta.yaml]: updated

File: recipes/hatch-ci/meta.yaml (hunk 4 above)
Comment (cav71) [reply recipes/hatch-ci/meta.yaml:19]: done

File: recipes/hatch-ci/meta.yaml (hunk 8 above)
Comment (cav71) [reply recipes/hatch-ci/meta.yaml]: done

**Pull request merged**